src/backend/egl/context.rs

```diff
@@ -46,86 +47,67 @@ impl EGLContext {
     ///
     /// - The context must be created from the system default EGL library (`dlopen("libEGL.so")`)
     /// - The `display`, `config`, and `context` must be valid for the lifetime of the returned context.
-    pub unsafe fn from_raw<L>(
+    pub unsafe fn from_raw(
         display: *const c_void,
         config_id: *const c_void,
         context: *const c_void,
-        log: L,
-    ) -> Result<EGLContext, Error>
-    where
-        L: Into<Option<::slog::Logger>>,
-    {
+    ) -> Result<EGLContext, Error> {
         assert!(!display.is_null(), "EGLDisplay pointer is null");
         assert!(!config_id.is_null(), "EGL configuration id pointer is null");
         assert!(!context.is_null(), "EGLContext pointer is null");
-        let log = crate::slog_or_fallback(log.into()).new(o!("smithay_module" => "backend_egl"));
 
-        let display = EGLDisplay::from_raw(display, config_id, log)?;
+        let display = EGLDisplay::from_raw(display, config_id)?;
         let pixel_format = display.get_pixel_format(config_id)?;
 
+        let span = info_span!(parent: &display.span, "egl_context");
+
         Ok(EGLContext {
             context,
             display,
             config_id,
             pixel_format: Some(pixel_format),
             user_data: Arc::new(UserDataMap::default()),
             externally_managed: true,
+            span,
         })
     }
 
     /// Creates a new configless `EGLContext` from a given `EGLDisplay`
-    pub fn new<L>(display: &EGLDisplay, log: L) -> Result<EGLContext, Error>
-    where
-        L: Into<Option<::slog::Logger>>,
-    {
-        Self::new_internal(display, None, None, log)
+    pub fn new(display: &EGLDisplay) -> Result<EGLContext, Error> {
+        Self::new_internal(display, None, None)
     }
 
     /// Create a new [`EGLContext`] from a given `EGLDisplay` and configuration requirements
-    pub fn new_with_config<L>(
+    pub fn new_with_config(
         display: &EGLDisplay,
         attributes: GlAttributes,
         reqs: PixelFormatRequirements,
-        log: L,
-    ) -> Result<EGLContext, Error>
-    where
-        L: Into<Option<::slog::Logger>>,
-    {
-        Self::new_internal(display, None, Some((attributes, reqs)), log)
+    ) -> Result<EGLContext, Error> {
+        Self::new_internal(display, None, Some((attributes, reqs)))
     }
 
     /// Create a new configless `EGLContext` from a given `EGLDisplay` sharing resources with another context
-    pub fn new_shared<L>(display: &EGLDisplay, share: &EGLContext, log: L) -> Result<EGLContext, Error>
-    where
-        L: Into<Option<::slog::Logger>>,
-    {
-        Self::new_internal(display, Some(share), None, log)
+    pub fn new_shared(display: &EGLDisplay, share: &EGLContext) -> Result<EGLContext, Error> {
+        Self::new_internal(display, Some(share), None)
```

Comment:
It might be useful to store the context from which objects are being shared in the span?